packages/insured-bridge-relayer/src/RelayerHelpers.ts

```diff
@@ -35,17 +35,20 @@ export async function approveL1Tokens(
   }
 }
 
-// Iterate over provided list of whitelisted L1 tokens and remove any that are not whitelisted on a specific L2.
+// Iterate over queried list of whitelisted L1 tokens and remove any that are not whitelisted on a specific L2.
 // The Relayer will use this whitelist to determine which deposits to relay. By default, the whitelist will be set
-// to the list of all possible tokens provided in the RateModels dictionary. However, not all L1 tokens in this
+// to the list of all possible tokens provided in the on-chain RateModels dictionary. However, not all L1 tokens in this
 // dictionary will be whitelisted for all L2 deposit boxes.
 export async function pruneWhitelistedL1Tokens(
   logger: winston.Logger,
   l1Client: InsuredBridgeL1Client,
-  l2Client: InsuredBridgeL2Client,
-  whitelistedRelayL1Tokens: string[]
+  l2Client: InsuredBridgeL2Client
 ): Promise<string[]> {
   await Promise.all([l2Client.update(), l1Client.update()]);
+
+  // Fetch list of potential whitelisted L1 tokens from keys in the RateModelStore.
+  const whitelistedRelayL1Tokens = l1Client.getL1TokensFromRateModel();
```

Comment:
Do we care (or is there anything we should do) if the rate model doesn't include a token that's whitelisted by the bridgeAdmin?

Do you think this is something we should error or warn on?

Reply:
Hm that means we need to iterate across ALL whitelisted L1 tokens on the `bridgeAdmin` which is kind of hard to do unless we maintain a list of whitelisted tokens right? I suppose we do store all whitelisted L1 tokens in the `bridgePoolDeployData` struct in `RelayerConfig.ts`

Reply:
Isn't there an event we can parse in the BridgeAdmin to grab all of the whitelisted tokens?

Not saying we _have_ to do this, just pointing out that we're using the rate model as the source of truth, but there's a similarly valid source of truth in the BridgeAdmin whitelist. Not sure if we should try to detect when those are different from one another. I could see it being fine for there to be a rate model for a token that isn't whitelisted, but a whitelisted token with no rate model seems odd at best and a security issue at worst.

Thoughts?

Reply:
Ah you're right we totally can use `l1Client.getWhitelistedTokensForChainId` instead of calling `bridgeAdmin.methods.whitelistToken()` directly. I'll fix this